persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java

```diff
@@ -299,65 +332,61 @@ public void write(MarshallableEntry entry) throws PersistenceException {
       Object key = getKey(entry);
       Object value = getValue(entry);
 
-      remoteCache.put(key, value, toSeconds(lifespan, entry.getKey(), LIFESPAN), TimeUnit.SECONDS,
-            toSeconds(maxIdle, entry.getKey(), MAXIDLE), TimeUnit.SECONDS);
+      return remoteCache.putAsync(key, value, toSeconds(lifespan, entry.getKey(), LIFESPAN), TimeUnit.SECONDS,
```

Comment:
To improve readability we can just do `long lifespan = metadata != null ? toSeconds(metadata.lifespan(), entry.getKey()) : -1;`

The put call is then just:

```java
      return remoteCache.putAsync(key, value, lifespan, TimeUnit.SECONDS, maxIdle, TimeUnit.SECONDS)
            .thenApply(CompletableFutures.toNullFunction());
```

I'm also not sure that the current trace message of outputting the key is particularly useful, so we could simplify `toSeconds` by removing the key param.

Reply:
Sure, I didn't think to reduce the existing code :)